packages/token-sdk/src/token-fetcher.ts

```diff
@@ -40,7 +50,7 @@ export class TokenFetcher {
       const mintInfo = await getParsedAccount(this.connection, mint, ParsableMintInfo);
```

Comment:
Do we need not to include in ``find`` the points added to findMany that makes isPartial a miss decision, or the change to use ptimeout?

How about making the implementation of find use findMany like the relationship between resolveOrCreateAtas and resolveOrCreateAta? (I think it would be difficult to see a clear performance difference between provider.find and provider.findMany.)

Reply:
That's a good suggestion, but I think it may sense to keep the `find` and `findMany` implementations distinct in the `TokenFetcher`.

This is the primary place where `MetadataProvider` will be used, and that interface also has `find` and `findMany`. Some metadata providers have separate implementations for `find` and `findMany` - for example, SolanaFM has different endpoints for single and batch calls. CoinGecko, on the other hand, only has a single endpoint, and the `findMany` batches multiple `find` calls.

If the `TokenFetcher` uses `findMany` to implement `find`, it may be confusing because users will assume `MetadataProvider#find` is being used. It also makes sense to keep the `MetadataProvider` interface as is because of the different client implementations as I mentioned.

Reply:
Regarding your other comments - 

- the `isPartial` check is in the method, it's lower inside the `for` loop where it breaks if all metadata is found.
- I added the `pTimeout` call as well as added a `mergeEntry` method to re-use more code between `find` and `findMany`